lib/Feature/FeatureModelTransaction.cpp

```diff
@@ -50,46 +52,82 @@ void removeFeatures(FeatureModel &FM,
           Overloaded{[&ActualFeature, &FM](Feature *F) { ActualFeature = F; },
                      [&ActualFeature, &FM](string &F) {
                        ActualFeature = FM.getFeature(F);
                      }},
-          *FeatureIterator);
+          RemainingFeature);
       // if Feature is not a parent it can be deleted
       if (ActualFeature->children().empty()) {
-        Trans.removeFeature(*FeatureIterator, Recursive);
-        DeleteFeatures.emplace_back(*FeatureIterator);
+        Trans.removeFeature(RemainingFeature, Recursive);
+        DeleteFeatures.emplace_back(RemainingFeature);
       }
     }
     // We are not in recursive mode and have only parents left with children
     // that should not be deleted --> abort execution
     if (DeleteFeatures.empty()) {
       Trans.commit();
       if (Recursive) {
-        removeFeaturesRecursively(FM, std::move(RemainingFeatures));
+        std::sort(RemainingFeatures.begin(), RemainingFeatures.end(),
+                  [&FM](detail::FeatureVariantTy &FV1,
+                        detail::FeatureVariantTy &FV2) -> bool {
+                    Feature *ActualFeature1;
+                    std::visit(Overloaded{[&ActualFeature1, &FM](Feature *F) {
+                                            ActualFeature1 = F;
+                                          },
+                                          [&ActualFeature1, &FM](string &F) {
+                                            ActualFeature1 = FM.getFeature(F);
+                                          }},
+                               FV1);
+                    Feature *ActualFeature2;
+                    std::visit(Overloaded{[&ActualFeature2, &FM](Feature *F) {
+                                            ActualFeature2 = F;
+                                          },
+                                          [&ActualFeature2, &FM](string &F) {
+                                            ActualFeature2 = FM.getFeature(F);
+                                          }},
+                               FV2);
+
+                    return countNumberOfSuccessors(ActualFeature1) <
+                           countNumberOfSuccessors(ActualFeature2);
+                  });
+        removeFeaturesRecursively(FM, RemainingFeatures);
       }
       // TODO: return list of non-deletable Features?
       return;
     }
+
     std::sort(DeleteFeatures.begin(), DeleteFeatures.end());
     RemainingFeatures.erase(
         std::remove_if(RemainingFeatures.begin(), RemainingFeatures.end(),
-                       [&](auto x) {
+                       [&](auto X) {
                          return binary_search(DeleteFeatures.begin(),
-                                              DeleteFeatures.end(), x);
+                                              DeleteFeatures.end(), X);
                        }),
         RemainingFeatures.end());
     Trans.commit();
   }
 }
 
+int countNumberOfSuccessors(FeatureTreeNode *F) {
```

Comment:
maybe we should integrate this into the FeatureTreeNode class or at least on the corresponding header